test/fail_compilation/fail20638.d

```diff
@@ -0,0 +1,13 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail20638.d(12): Error: undefined identifier `foo` in module `imports.fail20638b`
```

Comment:
Surely the compiler can know that it exists and is private and tell the users that.

Reply:
Agreed. Separate issue though - by the time it gets to corrections, it's already too late.

Reply:
Really? Surely
```d
if (s && symbolIsVisible(sc, s))
{
    if (s.isPackage()) { ... }
    else { /*error with symbol hint*/ }
}
else if (s) { /*error symbol is private*/ }
else { /* error no symbol found*/ }
```

Reply:
No, because `s` is the spelling-corrected symbol, not the original.

Reply:
Ah.